### PR TITLE
Feat - 269 - Added universities and points of contact to applicants

### PR DESCRIPTION
**Pull request by beatriz-sanchez-wizeline**
# What does this PR do?

This PR adds the Universities and UniversityPointsOfContact tables and corresponding models. For each there is a view in the admin panel where admins can add or edit universities and points of contacts. An importing script is also provided in case universities and points of contacts come from a csv file.

Additionally, applicants were updated to use the university and university contact info in their application. Views where interns are listed or shown have been updated to use the university and point of contact info from the db.

# Any background context you want to provide?

For the importing scripts, I came across several issues when inserting multiple records (with relations) to the db within an async loop. The solution for this is retrying according to the [prisma documentation](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#unique-key-constraint-errors-on-upserts).

**To get a copy of the universities and points of contact csv please write in the internal slack channel.**

# Where should the reviewer start?

1. `prisma/schema.prisma`
2. `tasks/updateUniversitiesAndContacts.ts` 
3. `tasks/updateApplicants.ts`
4. Then look at the model and route files.

# How should this be manually tested?

## Setup

DB setup
```
npx prisma migrate reset
npx prisma migrate deploy
npx prisma generate
npm run dev # Rerun to ensure it picks up the db changes
```

Insert universities and points of contact
```
npx ts-node -r dotenv/config tasks/updateUniversitiesAndContacts.ts [location of universities.csv file]
```

Insert applicants
```
npx ts-node -r dotenv/config tasks/updateApplicants.ts [location of interns.csv file]
```

## Steps

**As an intern**
1. Log in as a new intern
2. Fill in the form

**As Admin**
1. Log in as Wizeliner admin
2. Go to the admin panel and look for the Universities tab and all its sub tabs.
5. Go to interns applicant view
6. Choose an applicant and view its details

# What are the relevant tickets?

#269 
#270 
#271 

# Screenshots

## Admin Panel: Universities
Screenshot 2023-11-23 at 11 13 57
Screenshot 2023-11-23 at 11 14 07
Screenshot 2023-11-23 at 11 14 15

## Admin Panel: Points of Contact
Screenshot 2023-11-23 at 11 14 23
Screenshot 2023-11-23 at 11 14 35
Screenshot 2023-11-23 at 11 14 28

## Applicants List
Screenshot 2023-11-23 at 11 13 40

## Applicant View
Screenshot 2023-11-23 at 11 10 15

## Application Form

https://github.com/wizeline/remix-project-lab/assets/149706555/52ed375e-5db8-4cfd-b555-7b10ff3382e4

# Questions

N/A

# Checklist

- [x] I added the necessary documentation, if appropriate.
- [ ] I added tests to prove that my fix is effective or my feature works.
- [x] I reviewed existing Pull Requests before submitting mine.

### CODE REVIEW

**Comment by martinrobled0** [comment]
I was able to create a point of contact without selecting a university.
image

File: app/routes/manager/universities/points-of-contact.tsx [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+}
+
+export const validator = withZod(
+  z.object({
+    id: z.string().optional(),
+    fullName: z
+      .string()
+      .min(1, { message: "Name is required" }),
+    university: z.object({
+      name: z.string()
```
Comment (AxelUZ) [inline app/routes/manager/universities/points-of-contact.tsx:40]: I have the same problem as Martin, maybe here is the problem.

**Comment by martinrobled0** [comment]
Also I thinking that should be great that we can disable some universities in case that the relation finished, but not sure if this could happen.
Could happen @jackbravo ?

**Comment by beatriz-sanchez-wizeline** [comment]
> I was able to create a point of contact without selecting a university. image

Well spotted, I'll make sure this is properly validated.

> Also I thinking that should be great that we can disable some universities in case that the relation finished, but not sure if this could happen. Could happen @jackbravo ?

I think it is possible, I had added an active property to both universities and contacts in the DB but I wasn't entirely convinced. However I think this is better than deleting the universities and/or contacts as it could leave interns without universities. I'll add a set Active / Inactive button to both universities and contacts.

**Comment by beatriz-sanchez-wizeline** [comment]
Added the active property to universities and points of contacts. Visible on the index view and on the modify modal. For the creation I assume that we want both universities and contacts to be active at first.

Let me know your thoughts on the UI (below). I think that the table needs updating so that sorting and filtering is possible. But I would suggest doing this in another branch. 

Validation of point of contact is still broken. I'll provide an update later.

## Universities
Screenshot 2023-11-24 at 15 04 54
Screenshot 2023-11-24 at 15 05 05

## Points of Contact
Screenshot 2023-11-24 at 15 04 30
Screenshot 2023-11-24 at 15 04 39

File: app/routes/manager/universities/points-of-contact.tsx [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+}
+
+export const validator = withZod(
+  z.object({
+    id: z.string().optional(),
+    fullName: z
+      .string()
+      .min(1, { message: "Name is required" }),
+    university: z.object({
+      name: z.string(),
```
Comment (AxelUZ) [inline app/routes/manager/universities/points-of-contact.tsx:43]: Adding a .min(1) to the university name on line 43 might work.

Comment (beatriz-sanchez-wizeline) [reply app/routes/manager/universities/points-of-contact.tsx]: Added :) Thanks for the suggestion

**Comment by beatriz-sanchez-wizeline** [comment]
Updated active/inactive view on both universities and points of contact as follows
Screenshot 2023-11-28 at 9 53 53

**Review by AxelUZ — APPROVED** [review]
I like the way the Active/Inactive status is displayed and the error of adding a contact point without a university has been corrected. LGTM! 👍

**Pull request merged**